w-dms/sqliteDriver.go

```diff
@@ -0,0 +1,279 @@
+		return d.readByCondition("where t.alias = ?", alias)
+	}
+	return d.readByCondition("")
+}
+
+func (d *sqliteDriver) Update(t Task) error {
+	var canCommit = false
+	transaction, err := d.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if canCommit {
+			transaction.Commit()
+		} else {
+			transaction.Rollback()
+		}
+	}()
+	if err = cleanTagsAndRemindersInTransaction(t, transaction); err != nil {
+		return err
+	}
+	typeID, err := d.findTypeID(t.Type)
+	if err != nil {
+		return err
+	}
+	res, err := transaction.Exec(updateClause, t.Alias, t.Description, typeID, t.Timestamp, t.EstimateTime, t.RealTime, t.ID)
+	if err != nil {
+		return err
+	}
+	count, err := res.RowsAffected()
+	if err == sql.ErrNoRows || count == 0 {
+		return errTaskNotFound
+	} else if err != nil {
+		return err
+	}
+	if err = addTaskTagsAndRemindersInTransaction(t, transaction); err != nil {
+		return err
+	}
+	canCommit = true
+	return nil
+}
+
+func (d *sqliteDriver) Delete(t Task) error {
+	id, err := strconv.ParseInt(t.ID.(string), 10, 0)
+	if err != nil {
+		return err
+	}
+	t.ID = id
+	var canCommit = false
+	transaction, err := d.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if canCommit {
+			transaction.Commit()
+		} else {
+			transaction.Rollback()
+		}
+	}()
+	if err = cleanTagsAndRemindersInTransaction(t, transaction); err != nil {
+		return err
+	}
+	res, err := transaction.Exec("delete from Tasks where id = ?", t.ID)
+	if err != nil {
+		return err
+	}
+	count, err := res.RowsAffected()
+	if err == sql.ErrNoRows || count == 0 {
+		return errTaskNotFound
+	} else if err != nil {
+		return err
+	}
+	canCommit = true
+	return nil
+}
+
+func (d *sqliteDriver) findTypeID(typeName string) (int, error) {
+	row := d.db.QueryRow("select id from Types where type = ?", typeName)
+	var id int
+	err := row.Scan(&id)
+	if err != nil {
+		return 0, err
+	}
+	return id, nil
+}
+
+func addTaskTagsAndRemindersInTransaction(t Task, tx *sql.Tx) error {
+	var tagStm, remStm *sql.Stmt
+	defer func() {
+		if tagStm != nil {
+			tagStm.Close()
+		}
+		if remStm != nil {
+			remStm.Close()
+		}
+	}()
+	tagStm, err := tx.Prepare("insert into Tags(task_id, tag) values(?, ?)")
+	if err != nil {
+		return err
+	}
+	for _, tag := range t.Tags {
+		_, err = tagStm.Exec(t.ID, tag)
+		if err != nil {
+			return err
+		}
+	}
+	remStm, err = tx.Prepare("insert into Reminders(task_id, reminder) values(?, ?)")
+	if err != nil {
+		return err
+	}
+	for _, reminder := range t.Reminders {
+		_, err = remStm.Exec(t.ID, reminder)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func cleanTagsAndRemindersInTransaction(task Task, tx *sql.Tx) error {
+	_, err := tx.Exec("delete from Tags where task_id = ?", task.ID)
+	if err != nil {
+		return err
+	}
+	_, err = tx.Exec("delete from Reminders where task_id = ?", task.ID)
+	return err
+}
+
+func (d *sqliteDriver) readByCondition(condition string, args ...interface{}) (TaskList, error) {
```

Comment:
It's just my opinion, but you could change signature to `func(r *sql.Rows, err error) (TaskList, error)`.
That way you would decouple function from sql query (`selectCaluse`), get rid of ugly arguments `args ...interface{}` and string concatenation.